### ISSUE TITLE
bug(sink): json encoder does not preserve microsecond precision

### ISSUE THREAD

**Issue by xiangjinwu** [risingwavelabs/risingwave#13564]

**Comment by xuefengze** [comment]
It seems that the `time` type ignores the millisecond part  in [json encoder](https://github.com/risingwavelabs/risingwave/blob/main/src/connector/src/sink/encoder/json.rs#L237)?

**Comment by xiangjinwu** [comment]
> It seems that the `time` type ignores the millisecond part in [json encoder](https://github.com/risingwavelabs/risingwave/blob/main/src/connector/src/sink/encoder/json.rs#L237)?

Yes, and we need `TimeHandlingMode` to fix it without breaking backward compatibility.

**Comment by tabVersion** [comment]
> > It seems that the `time` type ignores the millisecond part in [json encoder](https://github.com/risingwavelabs/risingwave/blob/main/src/connector/src/sink/encoder/json.rs?rgh-link-date=2023-11-24T08%3A05%3A27Z#L237)?
> 
> Yes, and we need `TimeHandlingMode` to fix it without breaking backward compatibility.

Can we resue the same options as the json parser in source?

**Comment by xiangjinwu** [comment]
> > > It seems that the `time` type ignores the millisecond part in [json encoder](https://github.com/risingwavelabs/risingwave/blob/main/src/connector/src/sink/encoder/json.rs?rgh-link-date=2023-11-24T08%3A05%3A27Z#L237)?
> > 
> > 
> > Yes, and we need `TimeHandlingMode` to fix it without breaking backward compatibility.
> 
> Can we resue the same options as the json parser in source?

Yes for the enum definition
https://github.com/risingwavelabs/risingwave/blob/14ff124712a2dd18cc9294a692fcaa6cf77ae334/src/connector/src/parser/unified/json.rs#L41-L45

Additionally it may need to be exposed as a format option #13109
